### DISCUSSION TITLE
JSON validator - target parameter

### DISCUSSION THREAD

**Discussion by JohnNelsonCog** [CMSgov/price-transparency-guide-validator#48] [Q&A] [answered]
Hello, I have a question about the -t parameter when calling the validator on a json file. does "in-network-rates" parameter validate all sections of the file or just the data under the "In_network" segment? along with that question what does the -t parameter = "provider-reference" validate? We want to validate that it just validates the schema for the provider-references file, but does not validate the provider_references section in the in-network-rates files. 
Also, when we validate an InNetwork file without the location element:
{
    "provider_group_id": 2,
    "location": "https://www.cms.gov/some_provider_group.json"
  }],
it returns that it is not valid json, but if you add the location element to the same file, it comes back as valid json.
can someone please explain that too.
thank you.

**Comment by carrils** [comment, marked as answer]
Hello @JohnNelsonCog, let me try and help by answering some of your questions: 
1) When setting `-t` to `in-network-rates`, all sections of the file will be validated, not any subsection.
2) Setting the target to be `provider-reference` will try to validate a file against the provider reference schema (thus the file being validated should be a provider reference file)
3) In regards to your question about the location element, rules around "location" can be found here: https://github.com/CMSgov/price-transparency-guide/blob/fba9de788bd00232360fecda43cae1c8b5b1c0ed/schemas/in-network-rates/in-network-rates.json#L143-L146

**Comment by carrils** [comment]
Hi @JohnNelsonCog, did that help answer your questions?

**Reply by JohnNelsonCog** [reply]
yes, it did. thank you.

**Reply by carrils** [reply]
Great to hear! Have a good day.
